packages/mobile_app/lib/screens/registration/screen.dart

```diff
@@ -0,0 +1,226 @@
+                  hint: "mm",
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${text.length}/$min";
+                    if (int.parse(text) > 12 || int.parse(text) < 1) return "x";
+                    return null;
+                  }),
+              day: UserInfoField(
+                  textObject: day,
+                  maxLength: 2,
+                  keyboardType: TextInputType.datetime,
+                  hint: "dd",
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${text.length}/$min";
+                    if (int.parse(text) > 31 || int.parse(text) < 1) return "x";
+                    return null;
+                  }),
+              year: UserInfoField(
+                  textObject: year,
+                  maxLength: 4,
+                  keyboardType: TextInputType.datetime,
+                  hint: "yyyy",
+                  validator: (String text) {
+                    int min = 4;
+                    if (text.isEmpty || text.length < min)
+                      return "${text.length}/$min";
+                    return null;
+                  }),
+              sex: UserInfoPicker(
+                items: indexToSex,
+                indexObject: sex,
+              ),
+              country: UserInfoField(
+                  textObject: country,
+                  maxLength: 50,
+                  hint: lang.birthcountry,
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${lang.atLeastXsymbolsNeeded} $min";
+                    return null;
+                  }),
+              place: UserInfoField(
+                  textObject: place,
+                  maxLength: 100,
+                  hint: lang.birthplace,
+                  validator: (String text) {
+                    int min = 2;
+                    if (text.isEmpty || text.length < min)
+                      return "${lang.atLeastXsymbolsNeeded} $min";
+                    return null;
+                  }),
+            ),
+            AcceptTermsRow(terms: termsAccepted),
+            SizedBox(
+              height: 32,
+            ),
+            GradientFlatButton(
+              onPressed: () {
+                // @DEBUG
+                // print("Name ${name.wrapped}");
+                // print("Month: ${month.wrapped}");
+                // print("Day: ${day.wrapped}");
+                // print("Year: ${year.wrapped}");
+                // print("Sex: ${indexToSex[sex.wrapped]}");
+                // print("Country: ${country.wrapped}");
+                // print("Place: ${place.wrapped}");
+
+                if (name.wrapped.isEmpty) return;
```

Comment:
TODO:  add a message to the user, that they must enter data.

Reply:
Popup?